### PR TITLE
Update the left sidebar design

### PR DESCRIPTION
**Pull request by ma12-co**
WIP to update the left sidebar design a bit. 

- [ ] Update the button style as per new specs;
- [x] Use normal color border for search element;
- [x] Show border only if conversations are scrolled down;
- [x] Restyle the hover feedback for conversation;

### CODE REVIEW

**Comment by ma12-co** [comment]
Hide top border when conv list not scrolled

Peek 2021-01-12 09-43

**Comment by ma12-co** [comment]
Update: Dropped the new button style commit for now

File: src/components/LeftSidebar/LeftSidebar.vue [hunk 1]
```diff
@@ -33,70 +35,75 @@
...
+						:conversations-list="conversationsList"
+						:initialised-conversations="initialisedConversations"
+						:search-text="searchText"
+						@click-search-result="handleClickSearchResult"
+						@focus="setFocusedIndex" />
+				</li>
+				<template v-if="isSearching">
+					<template v-if="!listedConversationsLoading && searchResultsListedConversations.length > 0">
+						<Caption
+							:title="t('spreed', 'Listed conversations')" />
```
Comment (nickvergessen) [inline src/components/LeftSidebar/LeftSidebar.vue:56]: nop

File: src/components/LeftSidebar/LeftSidebar.vue [hunk 2]
```diff
@@ -450,6 +459,21 @@ export default {
 				this.isFetchingConversations = false
 			}
 		},
+
+		// Checks whether the conversations list is scrolled all the way to the top
+		// or not
+		handleScroll() {
+			console.debug('i run')
```
Comment (nickvergessen) [inline src/components/LeftSidebar/LeftSidebar.vue:466]: you better do: 🏃

File: src/components/LeftSidebar/LeftSidebar.vue [hunk 3]
```diff
@@ -450,6 +459,20 @@ export default {
...
+
+		// Checks whether the conversations list is scrolled all the way to the top
+		// or not
+		handleScroll() {
+			const scrollTop = this.$refs.scroller.scrollTop
+			if (scrollTop === 0) {
+				this.isScrolledToTop = true
+			} else {
+				this.isScrolledToTop = false
+			}
```
Comment (nickvergessen) [inline src/components/LeftSidebar/LeftSidebar.vue:471]:
```suggestion
			this.isScrolledToTop = this.$refs.scroller.scrollTop === 0
```

**Review by nickvergessen — APPROVED** [review]

**Comment by nickvergessen** [comment]
Should we have waited until other apps also take this design approach?

**Comment by jancborchardt** [comment]
> Should we have waited until other apps also take this design approach?

Yeah, I also thought this was a WIP and was going to go into the Vue components. Otherwise we have more design disparity, like e.g. with the menus and such like we talked about @ma12-co

**Comment by ma12-co** [comment]
> Should we have waited until other apps also take this design approach?

I assume you're both talking about the border radius change on the conversation component? If so, it's just a border radius change, so shouldn't take long? 

@jancborchardt as far as the component goes, we still don't have one. 
Other than that, we spoke about this and I'm pretty sure I had your ok to go ahead with it in talk.

Anyway should I revert? Otherwise bring these changes to mail and contacts myself while we still don't have the component.

**Comment by nickvergessen** [comment]
Well my biggest problem is inconsistency with server. Maybe it's there in the files and settings app for 22, maybe not. But I prefer a consistent look across all parts

**Comment by jancborchardt** [comment]
> I assume you're both talking about the border radius change on the conversation component? If so, it's just a border radius change, so shouldn't take long?

Yes, so could you make sure that is consistent? E.g. this would then need to be done in the AppNavigation component for all apps, as well as in the AppContentList for Mail and Contacts. cc @skjnldsv

**Comment by ma12-co** [comment]
Sure @jancborchardt

**Comment by nickvergessen** [comment]
I created a list at https://github.com/nextcloud/spreed/issues/5240 with things that should go upstream

**Pull request merged**